src/Symfony/Component/HttpFoundation/Request.php

```diff
@@ -894,6 +898,24 @@ public function getBasePath()
      * @return string The raw URL (i.e. not urldecoded)
      */
     public function getBaseUrl()
+    {
+        $trustedPrefix = '';
+
+        // the proxy prefix must be prepended to any prefix being needed at the webserver level
+        if ($this->isFromTrustedProxy() && $trustedPrefix = $this->getTrustedValues(self::HEADER_X_FORWARDED_PREFIX)) {
```

Comment:
The error in tests is because of an `Array to string conversion` notice at line 909. This is caused by this assignment, where you override the `$trustedPrefix` variable with an empty array when the request is from a trusted proxy but without any trusted value for the prefix.
The solution is simple: use a different variable name for the array of trusted values.

Reply:
I've added a fix and also a test for empty values. However this does not seem to help, integration is still failing. 
Locally I dont have such errors. Could you have a second look?

Reply:
The tests which failed previously are now passing. The remaining failure is unrelated to your PR and already fails in master (this is being solved in https://github.com/symfony/symfony/pull/37728)

Reply:
ok thank you for clarifying that!